dev/com.ibm.ws.ras.instrument_test/test/com/ibm/ws/ras/RasTransformTest.java

```diff
@@ -206,83 +308,127 @@ public void testJava8StaticInitLambdasWithInnerClass() {
 
 			
 		} catch (Exception e) {
-			System.out.println("Exception: " + e.getMessage() + " " + unrollStack(e.getStackTrace()));
+			e.printStackTrace();
+			fail(e.getMessage());
 		}
 		
 	}
 	
-	//@Test
+	@Test
 	public void testJava8LambdaInjection() {
+		assumeTrue(Integer.parseInt(System.getProperty("java.specification.version")) >= 8);
+		String classPath = "test/test data/com/ibm/example/bytecode/HelloWorldJava8Lambdas.class";
+		assertTrue(assertByteCodeVersion(classPath, majorCodeMap.get("8")));
+		
 		try {
 			String classUnderTest = "com.ibm.example.bytecode.HelloWorldJava8Lambdas";
-			Class transformedClass = transformClass(classUnderTest, "test/test data/com/ibm/example/bytecode/HelloWorldJava8Lambdas.class");
+			Class<?> transformedClass = transformClass(classUnderTest, classPath);
 			
 			transformedClass.newInstance();		
 			
 			
 		} catch (Exception e) {
-			System.out.println("Exception: " +unrollStack(e.getStackTrace()));
+			e.printStackTrace();
+			fail(e.getMessage());
 		}
 		
 	}
 	
-
+	@Test //TODO expand this test to actually test Java 11 features https://openjdk.org/jeps/323
+	public void testJava11Basic() {
+		assumeTrue(Integer.parseInt(System.getProperty("java.specification.version")) >= 11);
+		String classPath = "test/test data/com/ibm/example/bytecode/HelloWorldJava11.class";
+		assertTrue(assertByteCodeVersion(classPath, majorCodeMap.get("11")));
+		
+		try {
+			String classUnderTest = "com.ibm.example.bytecode.HelloWorldJava11";
+			Class<?> transformedClass = transformClass(classUnderTest, classPath);
+			
+			transformedClass.newInstance();		
+		} catch (Exception e) {
+			e.printStackTrace();
+			fail(e.getMessage());
+		}
+		
+	}
 	
-	private void writeStringToFile(String toWrite, String path) {
-		PrintWriter file;
+	@Test //TODO expand this test to actually test Java 17 features https://openjdk.org/jeps/409
+	public void testJava17Basic() {
+		assumeTrue(Integer.parseInt(System.getProperty("java.specification.version")) >= 17);
+		String classPath = "test/test data/com/ibm/example/bytecode/HelloWorldJava17.class";
+		assertTrue(assertByteCodeVersion(classPath, majorCodeMap.get("17")));
+		
 		try {
-			file = new PrintWriter(path);
-			file.println(toWrite);
-			file.close();
-		} catch (FileNotFoundException e) {
-			// TODO Auto-generated catch block
+			String classUnderTest = "com.ibm.example.bytecode.HelloWorldJava17";
+			Class<?> transformedClass = transformClass(classUnderTest, classPath);
+			
+			transformedClass.newInstance();		
+		} catch (Exception e) {
 			e.printStackTrace();
+			fail(e.getMessage());
 		}
+		
 	}
 	
-	private void writeBytesToFile(byte[] toWrite, String path) {
+	//FIXME - This test is currently failing because isTransformPossible returns false
+	//TODO expand this test to actually test Java 21 features https://openjdk.org/jeps/441
+	//@Test 
+	public void testJava21Basic() {
+		assumeTrue(Integer.parseInt(System.getProperty("java.specification.version")) >= 21);
```

Comment:
This test will also fail because we build and run unit tests with Java 17 today and Java 17 cannot run Java 21 byte code as you noted in this line.  If we wanted to keep the test and actually run it, we would have to remove the assertTrue and just return if not >= 21
```suggestion
		if (Integer.parseInt(System.getProperty("java.specification.version")) < 21) {
		    return;
		}
```

Reply:
assumeTrue does the same thing.  It will marked the test as skipped.

Reply:
`assumeTrue` vs `assertTrue`  Wow those are close to each other.  I still think we can uncomment the `@Test` then.